addons/mail/models/mail_thread.py

```diff
@@ -1762,6 +1762,13 @@ def _mail_find_partner_from_emails(self, emails, records=None, force_create=Fals
             followers = records.mapped('message_partner_ids')
         else:
             followers = self.env['res.partner']
+        if records and records._fields.get('company_id'):
```

Comment:
We can use field names as in a dict, allowing to remove the usage of reserved _fields
```suggestion
        if records and 'company_id' in records:
```

Reply:
Note for forward port: there is a method in mail ``_mail_get_company_field`` that can be used, not sure in which version we added it

Reply:
There was indeed a merge conflict, [so added it in,](https://github.com/odoo/odoo/pull/159108) and discovered a 3 year old bug that way, as tests suddenly started failing because it was trying to call `_mail_get_company_field` on a domain.

Created a separate PR to fix it in versions 15.0 up to 16.4: https://github.com/odoo/odoo/pull/159265